keps/sig-cli/2985-public-krm-functions-registry/README.md

```diff
@@ -0,0 +1,938 @@
+
+Trust can be established at the publisher level. Users can choose to trust a
+publisher and use the KRM functions provided by this publisher.
+
+Publisher information can be used to aggregate KRM functions. For example, I can
+query https://krm-fn-registry.sigs.k8s.io/catalog.yaml?publisher=kustomize,kubeflow
+to find all the KRM functions offered by Kustomize and KubeFlow.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a KRM functions user, I can browse the function registry website (e.g.
+https://krm-fn-registry.sigs.k8s.io) and search the KRM function by name (e.g.
+set-labels). And I can find everything including doc, examples, homepage,
+maintainers information about the function.
+
+#### Story 2
+
+As a KRM function user, I can find all KRM functions published by Example Co.
+The metadata for these functions can be aggregated and served by the website.
+
+#### Story 3
+
+As a kustomize user, I can use the KRM functions provided by a publisher as
+[a catalog with kustomize](https://github.com/kubernetes/enhancements/pull/2954/files#diff-da31478d2b3a925c17471e989c953539b508eed0aae19d624ad943d08f4dd910R414-R415).
+The `kustomiztion.yaml` file may look like:
+
+```
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+catalogs:
+- https://krm-fn-registry.sigs.k8s.io/catalog.yaml?publisher=kustomize
+resources:
+- input.yaml
+transformers:
+- ...
+```
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+Security might be a potential risk here. We have introduced the publisher
+concept to mitigate the risk.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Repo Location
+
+The ideal repo location is in [kubernetes-sigs](https://github.com/kubernetes-sigs).
+It should be sponsored by SIG-CLI.
+
+The repo name can be `KRM-functions`, `KRM-function-catalog`,
```

Comment:
not sure what the capitalization rules are.... could be that it needs to be all lower case.